src/components/Item/ItemPreview.js

```diff
@@ -62,28 +64,32 @@ const ItemPreview = ({
       : "Posted " + Math.round(diffDays / 365) + " years ago";
   };
 
-  const routeToItem = (itemId) => {
-    loadItemDetail(itemId, postings);
-    window.location.href =
-      window.location.origin + "/items" + "/item=" + itemId;
+  const routeToItem = async (itemId) => {
+    try {
+      await loadItemDetail(itemId);
+      history.push({
+        pathname: "/items/item=" + itemId,
+      });
+    } catch (e) {
+      console.log(error);
+      /* no redirect; snackbar gives error feedback */
```

Comment:
Could you `dispatch(displayError("Items could not be loaded. Try again."))` here